llrpConnector/src/main/java/io/vantiq/extsrc/llrpConnector/LLRPConnector.java

```diff
@@ -0,0 +1,1546 @@
+ * (Default 500 milliseconds).  Some readers may not report the accurate LastSeenTime
+ * and insteads fill it with the FirstSeenTime (use this one for storing).
+ *
+ * Note: Requires an Hostname/IP Address and Port to communicate with the RFID Reader.
+ *
+ * @author Mark Silva msilva@vantiq.com
+ *
+ */
+public class LLRPConnector  {
+    Logger              log  = LoggerFactory.getLogger(this.getClass().getCanonicalName());
+
+    // Valid logLevels
+    private static ArrayList<String> LogLevels = new ArrayList<String>(Arrays.asList("error", "warn", "info", "debug"));
+    final static String INFO = "info";
+    final static String WARN = "warn";
+    final static String ERROR = "error";
+    final static String DEBUG = "debug";
+    final static String TAG_DATA_EVENT_NAME = "tagData";
+    final static String READER_DATA_EVENT_NAME = "readerData";
+    final static String READER_STATUS_EVENT_NAME = "readerStatus";
+    final static String M_SUCCESS = "M_Success";  // String value for successful status
+    final static Boolean READER_ONLINE = false;   // Used to set 'readerOffLine' property
+    final static Boolean READER_OFFLINE = true;   // Used to set 'readerOffLine' property
+
+    // Used to reconnect if necessary
+    private String hostname = null;		    // Hostname/IP address of the reader
+    private String sourceName = null;       // Vantiq Source name connected to
+    private int readerPort = -1;			// Reader communication port
+    private int tagReadInterval = 500;		// Frequency for ending AISpec
+    private int logLevel = -1;              // Log level to send to Vantiq
+    private static int ROSPEC_ID = 1;		// ID used for collecting tags
+
+    private Socket connection;				// Socket connection to reader
+    private DataOutputStream out;			// Stream when writing to reader
+    private ReadThread rt = null;			// Reads LLRP msg and add to queue
+    private Thread msgThread = null;		// Process regular msgs from reader
+    private Thread tagThread = null;		// Process tag data msgs from reader
+    private Thread readerAliveThread = null;	// Identify when reader is offline
+    private LLRPConnectorCore llrpCore;
+    private Boolean readerOffline = true;	// Indicates the Reader is Offline
+    private Boolean vantiqSourceOffline = true;	// Indicates the connection to the Vantiq Source is Offline
+    Instant lastKeepAlive = Instant.now();	// Last point we rcvd from reader
+
+    /*
+     * RECONNECT_INTERVAL is the duration to wait to check if Reader is online.
+     * It may take 15-30 seconds to flush through stored messages on the reader
+     * since we do not turn off the reading of tags (stored on reader and
+     * processed on reconnect).
+     * The KEEPALIVE_INTERVAL is the periodic interval the reader will use
+     * to send a KEEPALIVE message.  If we don't receive a KEEPALIVE message
+     * within the RECONNECT_INTERVAL, we drop the connection and try to
+     * reconnect indefinitely.
+     */
```

Comment:
This seems to imply that the connector is actively connecting to the reader.  The readme seems to imply that the reader's in charge.  Again, seems like an overview section would be helpful to define the communication, structure, etc.

Reply:
Correct, the connector is in charge and connects to the reader or disconnects when the vantiq connection goes down.  This is done primarily to allow the buffering of tag reads to be done by the reader and not the connector.

Though, while the vantiq connection is up, the reader connection may go down with the connector.  The reader is configured to send a KEEPALIVE message every 5 seconds - if we miss it, then we close and reconnect.